### PR TITLE
[#166] Fixes NPE when attempting to read/write NULL value for a List field within a Struct

### PR DESCRIPTION
**Pull request by avirtuos**
[#166] Fixes NPE when attempting to read/write NULL value for a List field within a Struct

### CODE REVIEW

File: athena-federation-sdk/src/test/java/com/amazonaws/athena/connector/lambda/data/BlockTest.java [hunk 1]
```diff
@@ -865,6 +866,117 @@ public void ListOfStructsTest()
...
+        ArrowRecordBatch batch = actualSerDe.deserialize(blockOut.toByteArray());
+
+        /**
+         * Generate and write the block
+         */
+        Block actualBlock = actualAllocator.createBlock(actualSchema);
+        actualBlock.loadRecordBatch(batch);
+        batch.close();
+
+        for (
```
Comment (soojinj) [inline athena-federation-sdk/src/test/java/com/amazonaws/athena/connector/lambda/data/BlockTest.java:942]: nit put in one line

File: athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/data/FieldResolver.java [hunk 2]
```diff
@@ -48,12 +48,12 @@
         public Object getFieldValue(Field field, Object value)
         {
             Types.MinorType minorType = Types.getMinorTypeForArrowType(field.getType());
-            if (minorType == Types.MinorType.LIST) {
```
Comment (soojinj) [inline athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/data/FieldResolver.java:51]: just curious why change the order?

File: athena-federation-sdk/src/test/java/com/amazonaws/athena/connector/lambda/data/BlockTest.java (hunk 1 above)
Comment (soojinj) [reply athena-federation-sdk/src/test/java/com/amazonaws/athena/connector/lambda/data/BlockTest.java:942]: the for loop

Comment (avirtuos) [reply athena-federation-sdk/src/test/java/com/amazonaws/athena/connector/lambda/data/BlockTest.java]: yikes, auto-formatter 0wned me. Fixed now

File: athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/data/FieldResolver.java (hunk 2 above)
Comment (avirtuos) [reply athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/data/FieldResolver.java:51]: basically its impossible to get an iterator of this type from a Map so its never going to work. It was supposed to be reversed.

**Pull request merged**